src/SonarScanner.MSBuild.TFS.Classic/BinaryToXmlCoverageReportConverter.cs

```diff
@@ -114,259 +61,35 @@ public bool ConvertToXml(string inputFilePath, string outputFilePath)
-                if (File.Exists(toolPath))
-                {
-                    var version = TryGetVersionAsDouble(kvp.Key);
-
-                    if (!double.IsNaN(version))
-                    {
-                        versionPathMap[version] = toolPath;
-                    }
-                }
-            }
-            return versionPathMap;
-        }
-
-        /// <summary>
-        /// Attempts to convert the supplied version to a double.
-        /// Returns NaN if the value could not be converted
-        /// </summary>
-        private static double TryGetVersionAsDouble(string versionKey)
-        {
-            if (!double.TryParse(versionKey, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out double result))
-            {
-                result = double.NaN;
-            }
-            return result;
-        }
-
-        #endregion Code Coverage Tool path from registry
-
         // was internal
-        public static bool ConvertBinaryToXml(string converterExeFilePath, string inputBinaryFilePath, string outputXmlFilePath,
+        public static bool ConvertBinaryToXml(string inputBinaryFilePath, string outputXmlFilePath,
             ILogger logger)
         {
-            Debug.Assert(!string.IsNullOrEmpty(converterExeFilePath), "Expecting the conversion tool path to have been set");
-            Debug.Assert(File.Exists(converterExeFilePath), "Expecting the converter exe to exist: " + converterExeFilePath);
             Debug.Assert(Path.IsPathRooted(inputBinaryFilePath), "Expecting the input file name to be a full absolute path");
             Debug.Assert(File.Exists(inputBinaryFilePath), "Expecting the input file to exist: " + inputBinaryFilePath);
             Debug.Assert(Path.IsPathRooted(outputXmlFilePath), "Expecting the output file name to be a full absolute path");
 
-            var args = new List<string>
-            {
-                "analyze",
-                string.Format(CultureInfo.InvariantCulture, @"/output:{0}", outputXmlFilePath),
-                inputBinaryFilePath
-            };
-
-            var scannerArgs = new ProcessRunnerArguments(converterExeFilePath, false)
+            var util = new CoverageFileUtility();
+            try
             {
-                WorkingDirectory = Path.GetDirectoryName(outputXmlFilePath),
-                CmdLineArgs = args,
-                TimeoutInMilliseconds = ConversionTimeoutInMs
-            };
-
-            var runner = new ProcessRunner(logger);
-            var success = runner.Execute(scannerArgs);
-
-            if (success)
-            {
-                // Check the output file actually exists
-                if (!File.Exists(outputXmlFilePath))
-                {
-                    logger.LogError(Resources.CONV_ERROR_OutputFileNotFound, outputXmlFilePath);
-                    success = false;
-                }
+                using var dummy = new ApplicationCultureInfo(CultureInfo.InvariantCulture);
```

Comment:
There might be better options available. e.g., there might be an option to convert to some intermediate `CoverageXmlFile` type and do the XML serialization ourselves. But it seems the required `CoverageFile.ReadAsync` method is internal. It should be investigated nevertheless. e.g. There is `public CoverageData ReadCoverageFile(string path)` on the util, but that seems to be a death end.

Reply:
I would avoid doing custom logic as much as we can.